### PR TITLE
Change EliminateDeadInputComponentsPass to EliminateDeadIOComponentsPass

### PR DESCRIPTION
**Pull request by greg-lunarg**
To reflect processing of both Input and Output variables.

Also renamed files as needed.

### CODE REVIEW

**Comment by greg-lunarg** [comment]
@keenuts PTAL

**Comment by Keenuts** [comment]
> @Keenuts PTAL

looking now.

File: source/opt/eliminate_dead_io_components_pass.h [hunk 1]
```diff
@@ -26,10 +26,10 @@ namespace spvtools {
 namespace opt {
 
 // See optimizer.hpp for documentation.
-class EliminateDeadInputComponentsPass : public Pass {
+class EliminateDeadIOComponentsPass : public Pass {
  public:
-  explicit EliminateDeadInputComponentsPass(bool output_instead = false,
-                                            bool vertex_shader_only = true)
+  explicit EliminateDeadIOComponentsPass(bool output_instead = false,
```
Comment (Keenuts) [inline source/opt/eliminate_dead_io_components_pass.h:31]: Still up to replace the output_instead by the storage class?
(https://github.com/KhronosGroup/SPIRV-Tools/pull/4984#discussion_r1018960530)

Comment (greg-lunarg) [reply source/opt/eliminate_dead_io_components_pass.h:31]: Yes. Sorry! Realized as I was heading out yesterday that I forgot this. Came in quick on my holiday today to fix :)

**Comment by greg-lunarg** [comment]
@Keenuts I think I have addressed all your concerns. PTAL.

**Review by Keenuts — APPROVED** [review]
Thanks for the cleanup!

File: source/opt/eliminate_dead_io_components_pass.h (hunk 1 above)
Comment (Keenuts) [reply source/opt/eliminate_dead_io_components_pass.h]: Oh yes US, happy thanksgiving!😊

**Pull request merged**